### ISSUE TITLE
Jump to index in sliver world

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#81005]
@goderbauer and I were talking about https://github.com/flutter/flutter/issues/12319 recently and came up with the following idea:

* ScrollControllers can have a new method jumpToIndex. That method extrapolates from current information to get a new (arbitrary) scroll offset pixel position, but stores the index specified as well.
* Viewports, when they see this index is specified, pass both the pixel value and the index to their sliver children and clear the flag on the scroll position.
* Sliver children use the index if available rather than the pixel position, if they know how to use indexes (AppBars use the scroll offset and ignore the index). In their geometry they say how many index positions they consumed.
* The viewport subtracts how many index positions were consumed from one child before moving on to the next.

**Comment by RaphaelGoulmot** [comment]
I totally agree, but the problem is on all the list/grid available on flutter, i needed to jump at specify position to do lazy loading on some huge list, but whenever i need to know the index position and the visibles elements it become impossible.

Using the package [scrollable_positioned_list](https://pub.dev/packages/scrollable_positioned_list), it become kinda possible but not at 100%, if i do a lazy loading from the bottom of a list, with this package i will get some difficulty to keep my view on the old element without having some white flash whenever i will refresh the position.

Anyway, i hope that they will create a better listing widget/system for everything.

**Comment by chunhtai** [comment]
If we don't need animation, we can just set the child at the index to be a scroll offset 0.0 in the sliver. Even though it is wrong, it will be correct once user start scrolling

**Comment by lukehutch** [comment]
`SliverMultiBoxAdaptorElement` has a method `estimateMaxScrollOffset`, which takes parameters `firstIndex` and `lastIndex`. Couldn't this be called to estimate the scroll position of a child element with a specific key? If so, it would be great if this could be exposed in the API, because with this, you could scroll to the applicable position with any `ScollController` implementation.

**Comment by vizakenjack** [comment]
This is a really important issue for me. Currently, I use `ScrollablePositionedList`, but it has some issues, such as requiring a custom `ItemScrollController` instead of a `ScrollController`, lacking certain attributes like `keyboardDismissBehavior`, and experiencing problems with lazy loading, as mentioned by @RaphaelGoulmot.

I think you need to prioritize this issue.

**Comment by stuartmorgan** [comment]
I'm temporarily locking this issue since someone has unfortunately chosen to try to organize the community to brigade this issue with comments that [violate our issue guidelines](https://github.com/flutter/flutter/blob/master/docs/contributing/issue_hygiene/README.md#do-not-add-me-too-or-same-or-is-there-an-update-comments-to-issues-or-prs).

If you came here at the direction of the post that encouraged brigading, please read [this section](https://github.com/flutter/flutter/blob/master/docs/contributing/issue_hygiene/README.md#do-not-add-me-too-or-same-or-is-there-an-update-comments-to-issues-or-prs), [this section](https://github.com/flutter/flutter/blob/master/docs/contributing/issue_hygiene/README.md#escalating-an-issue-that-has-the-wrong-priority), and [this section](https://github.com/flutter/flutter/blob/master/docs/contributing/issue_hygiene/README.md#thumbs-up-reactions) of our issue tracker guidelines.

**Comment by just-doit** [comment]
As this issue is already over three years old, I had hoped that the problem would have been solved by now. Any news?

**Comment by zeymermontes** [comment]
Unfortunally [scrollable_positioned_list](https://pub.dev/packages/scrollable_positioned_list) has some performance issue with complex children.

**Comment by KiddoV** [comment]
Any update on this? Also found this package but has limitation https://github.com/marcglasberg/indexed_list_view
https://github.com/marcglasberg/indexed_list_view#limitations

**Comment by XuanTung95** [comment]
Implementing this feature would be very helpful because developers no longer need to rely on a third-party package just to  jump to index.
Another possible approach would be to provide an API that can calculate the offset of an item by its index, even if that item has not been laid out yet:
`final offset = controller.offsetOfIndex(index);`
Then:
`controller.jumpTo(offset);`
